service/matching/taskListManager.go

```diff
@@ -359,6 +359,26 @@ func (c *taskListManagerImpl) CancelPoller(pollerID string) {
 
 // Returns a batch of tasks from persistence starting form current read level.
 func (c *taskListManagerImpl) getTaskBatch() ([]*persistence.TaskInfo, error) {
+	result := make([]*persistence.TaskInfo, 0)
+	readLevel := c.taskAckManager.getReadLevel()
+	maxReadLevel := c.taskWriter.GetMaxReadLevel()
+	// when gap is large, query db multiple times to avoid query aborted caused by tombstone
+	for readLevel < maxReadLevel {
+		upper := readLevel + c.config.RangeSize
+		if upper > maxReadLevel {
+			upper = maxReadLevel
+		}
+		tasks, err := c.getTaskBatchWithRange(readLevel, upper)
```

Comment:
Is the query inclusive or exclusive of upper?

Reply:
inclusive of upper (exclusive of lower/readLevel)
I found unit tests are covered in existing TestAddThenConsumeActivities etc, so didn't add additional ones.

Reply:
Since this is adding more functionality, it's worth adding unit tests for eg. With different range, verify that the right number of queries is issued.

Reply:
Agree and addressed, thanks.